concent_api/core/payments/bankster.py

```diff
@@ -265,80 +241,135 @@ def settle_overdue_acceptances(
+                client=requestor_client,
+                ethereum_address=requestor_ethereum_address,
+            )
+        except IntegrityError:
+            requestor_client = Client.objects.get_or_create_full_clean(
+                public_key=requestor_public_key,
+            )
+            requestor_deposit_account = DepositAccount.objects.get_or_create_full_clean(
+                client=requestor_client,
+                ethereum_address=requestor_ethereum_address,
+            )
+
     # Bankster asks SCI about the amount of funds available in requestor's deposit.
     requestor_deposit_value = service.get_deposit_value(client_eth_address=requestor_ethereum_address)  # pylint: disable=no-value-for-parameter
 
-    # Concent defines time T0 equal to oldest payment_ts from passed SubtaskResultAccepted messages from
-    # subtask_results_accepted_list.
-    oldest_payments_ts = min(subtask_results_accepted.payment_ts for subtask_results_accepted in acceptances)
+    # Bankster begins a database transaction and puts a database lock on the DepositAccount object.
+    with transaction.atomic(using='control'):
+        DepositAccount.objects.select_for_update().get(
+            pk=requestor_deposit_account.pk
+        )
 
-    current_time = get_current_utc_timestamp()
+        # Bankster sums the amounts of all existing DepositClaims that have the same payer as the one being processed.
+        sum_of_existing_requestor_claims = DepositClaim.objects.filter(
+            payer_deposit_account=requestor_deposit_account
+        ).aggregate(
+            sum_of_existing_claims=Coalesce(Sum('amount'), 0)
+        )
 
-    # Concent gets list of transactions from payment API where timestamp >= T0.
-    list_of_transactions = service.get_list_of_payments(  # pylint: disable=no-value-for-parameter
-        requestor_eth_address=requestor_ethereum_address,
-        provider_eth_address=provider_ethereum_address,
-        payment_ts=oldest_payments_ts,
-        current_time=current_time,
-        transaction_type=TransactionType.BATCH,
-    )
+        assert sum_of_existing_requestor_claims['sum_of_existing_claims'] >= 0
 
-    # Concent defines time T1 equal to youngest timestamp from list of transactions.
-    if len(list_of_transactions) == 0:
-        t1_is_bigger_than_payments_ts = False
-    else:
-        youngest_transaction = max(transaction.closure_time for transaction in list_of_transactions)
-        # Concent checks if all passed SubtaskResultAccepted messages from subtask_results_accepted_list
-        # have payment_ts < T1
-        t1_is_bigger_than_payments_ts = any(
-            youngest_transaction > subtask_results_accepted.payment_ts for subtask_results_accepted in acceptances
+        # If the existing claims against requestor's deposit are greater or equal to his current deposit,
+        # we can't add a new claim.
+        if requestor_deposit_value <= sum_of_existing_requestor_claims['sum_of_existing_claims']:
```

Comment:
I would suggest changing the condition, so it sounds exactly like your comment above:
```suggestion
        if  sum_of_existing_requestor_claims['sum_of_existing_claims'] >=:
```

Reply:
@rwrzesien, where was that rule? ;)

Reply:
@rwrzesien , what I found is: https://github.com/golemfactory/concent/pull/929#discussion_r232908597 and it is contrary to what you have stated above: 
`if value > 0` is used, rather than `if 0 < value`